src/formats/fastq.rs

```diff
@@ -109,12 +109,7 @@ impl<'a> Iterator for FastqParser<'a> {
         }
         let mut qual = &buf[id2_end..qual_end - 1];
 
-        if (qual_end + 1 < buf.len()
-            && buf[qual_end] != b'@'
-            && buf[qual_end] != b'\r'
-            && buf[qual_end] != b'\n')
-            || (qual_end < buf.len() && buf[qual_end - 1] != b'\n')
-        {
+        if qual_end < buf.len() && buf[qual_end - 1] != b'\n' {
```

Comment:
@bovee sooo in theory it should not work for cases where the quality line is at the end of the file with no newline but there is a test for that (`test_fastq`) and it still passes..